src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -68,7 +69,18 @@ public KafkaMessageDrivenChannelAdapter(AbstractMessageListenerContainer<K, V> m
 		this.messageListenerContainer.setAutoStartup(false);
 	}
 
+	/**
+	 * Set the message converter; must be a {@link RecordMessageConverter}.
+	 * @param messageConverter the converter
+	 * @deprecated in favor of {@link #setRecordMessageConverter(RecordMessageConverter)}.
+	 */
+	@Deprecated
 	public void setMessageConverter(MessageConverter messageConverter) {
+		Assert.isTrue(messageConverter instanceof RecordMessageConverter);
+		this.listener.setMessageConverter((RecordMessageConverter) messageConverter);
+	}
+
+	public void setRecordMessageConverter(RecordMessageConverter messageConverter) {
```

Comment:
Doh. Missed that somehow.
Fine with me.

Do you plan to have a separate setter on the matter?